src/python/bot/fuzzers/mutator_plugin.py

```diff
@@ -0,0 +1,156 @@
+from system import environment
+from system import shell
+
+MUTATOR_PLUGINS_BUCKET_ENV_VAR = 'MUTATOR_PLUGINS_BUCKET'
+MUTATOR_SHARED_OBJECT_FILENAME = 'mutator-plugin.so'
+ARCHIVES_SUBDIR_NAME = 'archives'
+PLUGINS_SUBDIR_NAME = 'plugins'
+
+
+def _get_mutator_plugins_bucket_url():
+  """Returns the url of the mutator plugin's cloud storage bucket."""
+  return 'gs://%s' % environment.get_value(MUTATOR_PLUGINS_BUCKET_ENV_VAR)
+
+
+def _get_mutator_plugins_subdir(subdir):
+  """Returns the path of the subdirectory named |subdir| in
+  MUTATOR_PLUGINS_DIR."""
+  return os.path.join(environment.get_value('MUTATOR_PLUGINS_DIR'), subdir)
+
+
+def _get_mutator_plugins_archives_dir():
+  """Returns path of the archives subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(ARCHIVES_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_unpacked_dir():
+  """Returns path of the unpacked plugins subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(PLUGINS_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_from_bucket():
+  """Returns list of the mutator plugin archives in the mutator plugin storage
+  bucket."""
+  return storage.list_blobs(_get_mutator_plugins_bucket_url())
+
+
+def _download_mutator_plugin_archive(mutator_plugin_archive):
+  """Downloads the |mutator_plugin_archive| from the mutator plugin storage
+  bucket to the plugin archives directory. Returns the path archive was
+  downloaded to."""
+  file_path = os.path.join(_get_mutator_plugins_archives_dir(),
+                           mutator_plugin_archive)
+  url = '%s/%s' % (_get_mutator_plugins_bucket_url(), mutator_plugin_archive)
+  assert gsutil.GSUtilRunner().download_file(url, file_path)
+  return file_path
+
+
+def _unpack_mutator_plugin(mutator_plugin_archive_path):
+  """Unpacks |mutator_plugin_archive_path| in the unpacked plugins directory and
+  returns the path it was unpacked into."""
+  mutator_plugin_name = os.path.basename(
+      os.path.splitext(mutator_plugin_archive_path)[0])
+  unpacked_plugin_dir = os.path.join(_get_mutator_plugins_unpacked_dir(),
+                                     mutator_plugin_name)
+  archive.unpack(mutator_plugin_archive_path, unpacked_plugin_dir)
+  return unpacked_plugin_dir
+
+
+class PluginGetter(object):
+  """Class that gets a usable mutator plugin for |fuzzer_binary_name| from
+  GCS."""
+
+  def __init__(self, fuzzer_binary_name):
+    self.fuzzer_binary_name = fuzzer_binary_name
+    self.job_name = environment.get_value('JOB_NAME')
+    self.create_directories()
+
+  def create_directories(self):
+    """Creates directories needed to use mutator plugins."""
+    plugins_root_dir = environment.get_value('MUTATOR_PLUGINS_DIR')
+    assert plugins_root_dir
+    shell.remove_directory(plugins_root_dir, recreate=True)
+    os.mkdir(_get_mutator_plugins_archives_dir())
+    os.mkdir(_get_mutator_plugins_unpacked_dir())
+
+  @staticmethod
+  def _extract_name_from_archive(plugin_archive_filename):
```

Comment:
nit: style guide says not to use staticmethod: 

http://google.github.io/styleguide/pyguide.html#2174-decision